### DISCUSSION TITLE
[v4] Dynamic @utility

### DISCUSSION THREAD

**Discussion by diegohaz** [tailwindlabs/tailwindcss#14779] [Help] [answered]
Hi! Amazing work with v4.

Are there plans to support dynamic `@utility` classes, similar to `matchUtilities`, that accept both defined and arbitrary values?

Or is the idea to continue using JavaScript for those?

**Comment by adamwathan** [comment, marked as answer]
Hey @diegohaz! Yeah I really want to crack that but it's been a struggle to nail the syntax.

I published my best effort internally for the team last week, curious if you have any feedback:

https://gist.github.com/adamwathan/ebb0d76827c92c4b165a4baa1a6d28b8

**Reply by diegohaz** [reply]
Oh, great to hear it's in progress.

I think the syntax makes sense. It would also work for setting variables, right?

```css
@utility test-* {
  --test: value([length]);
}
```

The issue with Prettier is unfortunate, but I'm fine with the workaround. I'm assuming it wouldn't make sense to pass any theme variable without the `-*` suffix to `value` or `modifier`, so could it be optional?

I'm currently using Biome, which doesn't seem to have this issue.
